### PR TITLE
DOC: Update genfromtxt docs to use StringIO and u-strings

### LINKED ISSUE #8498: In Python 3, BytesIO input for numpy.genfromtxt does not work as the examples say
The examples provided in https://docs.scipy.org/doc/numpy/user/basics.io.genfromtxt.html cannot work in Python 3 due to a TypeError. I got the following error message.
```
>>> data = "1, 2, 3\n4, 5, 6"
>>> np.genfromtxt(BytesIO(data), delimiter=",")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' does not support the buffer interface
```

It turns out that the input string needs to be cast to a bytestring for it to work properly. I hope the manual page [Importing data with `genfromtxt`](https://docs.scipy.org/doc/numpy/user/basics.io.genfromtxt.html) can be updated so that it states this issue properly.
```
>>> data = "1, 2, 3\n4, 5, 6"
>>> np.genfromtxt(BytesIO(data.encode('utf-8')), delimiter=",")
array([[ 1.,  2.,  3.],
       [ 4.,  5.,  6.]])
```

### PR DESCRIPTION
**Pull request by eric-wieser**
We could also instead use b-strings everywhere, but encouraging unicode seems like the right thing to do.

Fixes #8498

### CODE REVIEW

File: doc/source/user/basics.io.genfromtxt.rst [hunk 1]
```diff
@@ -58,8 +58,8 @@ Quite often, a single character marks the separation between columns.  For
 example, comma-separated files (CSV) use a comma (``,``) or a semicolon
 (``;``) as delimiter::
 
-   >>> data = "1, 2, 3\n4, 5, 6"
-   >>> np.genfromtxt(BytesIO(data), delimiter=",")
+   >>> data = u"1, 2, 3\n4, 5, 6"
+   >>> np.genfromtxt(StringIO(data), delimiter=",")
```
Comment (juliantaylor) [inline doc/source/user/basics.io.genfromtxt.rst]: the delimiter should be an u string too

Comment (juliantaylor) [reply doc/source/user/basics.io.genfromtxt.rst]: hm though there are a lot, maybe it is unnecessary clutter

Comment (charris) [reply doc/source/user/basics.io.genfromtxt.rst]: The delimiter gets promoted to unicode in Python 2.7, so I think this is OK.
```
In [1]: from io import StringIO

In [2]: data = u"1, 2, 3\n4, 5, 6"

In [3]: import numpy as np

In [4]: np.genfromtxt(StringIO(data), delimiter=",")
Out[4]: 
array([[1., 2., 3.],
       [4., 5., 6.]])
```

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**